addons/hr_attendance/models/ir_websocket.py

```diff
@@ -0,0 +1,25 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class IrWebsocket(models.AbstractModel):
+    """Override to handle hr_employee specific features (presence in particular)."""
+
+    _inherit = "ir.websocket"
+
+    def _build_bus_channel_list(self, channels):
+        channels = list(channels)  # do not alter original list
+        employee_channel_ids = []
+        has_employee_public_access = self.env["hr.employee.public"].has_access("read")
+
+        for channel in list(channels):
```

Comment:
I guess it's a typo that you transform `channels` into a list twice?
```suggestion
        employee_channel_ids = []
        has_employee_public_access = self.env["hr.employee.public"].has_access("read")

        for channel in list(channels):  # do not alter original list
```